Classes/Outlet/OutletArgument.php

```diff
@@ -0,0 +1,219 @@
+ */
+
+
+use TYPO3\CMS\Extbase\Property\Exception\TargetNotFoundException;
+use TYPO3\CMS\Extbase\Utility\TypeHandlingUtility;
+use TYPO3\CMS\Extbase\Validation\Exception\NoSuchValidatorException;
+use TYPO3\CMS\Extbase\Validation\Validator\ConjunctionValidator;
+use TYPO3\CMS\Extbase\Validation\Validator\ValidatorInterface;
+use TYPO3\CMS\Extbase\Validation\ValidatorResolver;
+
+/**
+ *
+ * @api
+ */
+class OutletArgument
+{
+    /**
+     * @var \TYPO3\CMS\Extbase\Property\PropertyMapper
+     * @inject
+     */
+    protected $propertyMapper;
+
+    /**
+     * @var \TYPO3\CMS\Extbase\Mvc\Controller\MvcPropertyMappingConfiguration
+     * @inject
+     */
+    protected $propertyMappingConfiguration;
+
+    /**
+     * @var \TYPO3\CMS\Extbase\Validation\ValidatorResolver
+     * @inject
+     */
+    protected $validatorResolver;
+
+    /**
+     * Name of this argument
+     *
+     * @var string
+     */
+    protected $name = '';
+
+    /**
+     * Data type of this argument's value
+     *
+     * @var string
+     */
+    protected $dataType = null;
+
+    /**
+     * Actual value of this argument
+     *
+     * @var mixed
+     */
+    protected $value = null;
+
+    /**
+     * A custom validator, used supplementary to the base validation
+     *
+     * @var array<\TYPO3\CMS\Extbase\Validation\Validator\ValidatorInterface>
+     */
+    protected $validators = null;
+
+    /**
+     * The validation results. This can be asked if the argument has errors.
+     *
+     * @var \TYPO3\CMS\Extbase\Error\Result
+     */
+    protected $validationResults = null;
+
+    /**
+     * Creates a new instance of the OutletArgument based on the name and dataTape this argument will represent
+     *
+     * @param string $name Name of this argument
+     * @param string $dataType The data type of this argument
+     * @throws \InvalidArgumentException if $name is not a string or empty
+     */
+    public function __construct($name, $dataType)
+    {
+        if (!is_string($name)) {
+            throw new \InvalidArgumentException('$name must be of type string, ' . gettype($name) . ' given.', 1187951688);
+        }
+        if (empty($name)) {
+            throw new \InvalidArgumentException('$name must be a non-empty string.', 1232551853);
+        }
+        $this->name = $name;
+        $this->dataType = TypeHandlingUtility::normalizeType($dataType);
+    }
+
+    /**
+     * Returns the name of this argument
+     *
+     * @return string This argument's name
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Returns the data type of this argument's value
+     *
+     * @return string The data type
+     */
+    public function getDataType()
+    {
+        return $this->dataType;
+    }
+
+    /**
+     * Sets custom validators which are used supplementary to the base validation
+     *
+     * @param ValidatorInterface[] $validators The actual validator object
```

Comment:
Actually I like these more than the `array<type>` thing and it's well supported in any IDE. I propose you change the annotations for every array field to `type[]`